File: DataSets/DataSet-New/LeetCodeTasks - Develop/LeetCodeTasks/3127_make-a-square-with-the-same-color/solutions/java/solution.java
class Solution {
    public boolean canMakeSquare(char[][] grid) {
        // Check all 2x2 sub-squares in the 3x3 grid
        for (int i = 0; i < 2; i++) {
            for (int j = 0; j < 2; j++) {
                int bCount = 0;
                int wCount = 0;
                
                // Count B's and W's in the current 2x2 square
                for (int x = i; x <= i + 1; x++) {
                    for (int y = j; y <= j + 1; y++) {
                        if (grid[x][y] == 'B') bCount++;
                        else wCount++;
                    }
                }
                
                // Check if we can make all the same with at most one change
                if (bCount == 4 || wCount == 4 || bCount == 3 || wCount == 3) {
                    return true;
                }
            }
        }
        return false;
    }
}

File: DataSets/DataSet-New/LeetCodeTasks - Develop/LeetCodeTasks/3127_make-a-square-with-the-same-color/solutions/cpp/solution.cpp
#include <vector>

class Solution {
public:
    bool canMakeSquare(std::vector<std::vector<char>>& grid) {
        // Check all four possible 2x2 squares in the 3x3 grid
        for (int row = 0; row < 2; ++row) {
            for (int col = 0; col < 2; ++col) {
                // Count the number of 'B' and 'W' in the current 2x2 square
                int black = 0, white = 0;
                for (int i = 0; i < 2; ++i) {
                    for (int j = 0; j < 2; ++j) {
                        if (grid[row + i][col + j] == 'B') {
                            black++;
                        } else {
                            white++;
                        }
                    }
                }
                
                // If we can change at most one cell to make all same color
                if (black == 4 || white == 4 || black == 3 || white == 3) {
                    return true;
                }
            }
        }
        
        return false;
    }
};
